### DISCUSSION TITLE
Get internal asset graph or repository_def from AssetExecutionContext

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#19304] [Q&A]
Hi, I need to be able to access the InternalAssetGraph (i.e. via the repository_def), from inside an asset execution function, how can I do this?

**Comment by jamiedemaria** [comment]
It doesn't look like there's a way to get the `InternalAssetGraph` within an asset's execution function. You can get access to the `AssetLayer` via `context.job_def.asset_layer` though. That might have what you need

**Reply by the4thamigo-uk** [reply]
Is that something that you would consider as a feature request?

**Reply by the4thamigo-uk** [reply]
p.s. the asset_layer only seems to provide the assets within the job, but I would like to interrogate the graph for upstream assets outside of the job.

**Comment by the4thamigo-uk** [comment]
converted to an issue
